droid-api/src/main/java/uk/gov/nationalarchives/droid/internal/api/DroidAPI.java

```diff
@@ -243,18 +274,68 @@ private List<ApiResult> submitHttpIdentification(final URI uri, String extension
 
         final RequestIdentifier id = getRequestIdentifier(uri);
 
+        Map<HashAlgorithm, String> hashResults = generateHashResults(uri, this::getHttpHash);
 
         try (final HttpIdentificationRequest request = new HttpIdentificationRequest(metaData, id, httpClient)) {
             request.setExtension(extension);
             request.open(uri);
-            return getApiResults(request);
+            return List.of(new APIResult(getIdentificationResults(request), hashResults));
         }
     }
 
-    private List<ApiResult> submitS3Identification(final URI uri, String extension) throws IOException {
+    private <T> Map<HashAlgorithm, String> generateHashResults(T identifier, BiFunction<HashAlgorithm, T, String> hashFunction) {
+        return hashAlgorithms.stream().collect(Collectors.toMap(
+                algorithm -> algorithm,
+                algorithm -> hashFunction.apply(algorithm, identifier)
+        ));
+    }
+
+    private String getFileHash(HashAlgorithm hashAlgorithm, Path path) {
```

Comment:
Is the FileInputStream staying open here? 
Might be easier to use try-with-resource so it closes automatically 

```
        try (InputStream fs = new FileInputStream(path.toFile())){
            return getHash(hashAlgorithm, fs);
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
```

Reply:
Good point, I've changed it.